centrifuge-js/src/modules/pools.ts

```diff
@@ -1067,17 +1132,17 @@ export function getPoolsModule(inst: CentrifugeBase) {
       switchMap((api) => api.query.loans.loanInfo(poolId, loanId)),
       map((loanData) => {
         const loanValue = loanData.toJSON() as unknown as LoanDetailsData
-        const assetKey = (loanData.toHuman() as any).asset
+        const assetKey = (loanData.toHuman() as unknown as LoanDetailsData).collateral
 
         const loan: Loan = {
           id: loanId,
           poolId,
-          financedAmount: parseBN(loanValue.borrowedAmount),
-          financingFee: parseBN(loanValue.ratePerSec),
-          outstandingDebt: new BN(parseBN(loanValue.principalDebt))
-            .mul(new BN(parseBN(loanValue.accumulatedRate)))
-            .div(new BN(10).pow(new BN(27)))
-            .toString(),
+          financingFee: new Rate(hexToBN(loanValue.interestRatePerSec)),
```

Comment:
We should rename this to:

```suggestion
          interestRatePerSec: new Rate(hexToBN(loanValue.interestRatePerSec)),
```